### ISSUE TITLE
undefined method `url' for #<Datadog::Transport::HTTP::Adapters::Test:0x00007fde4b256bc0>

### ISSUE THREAD

**Issue by cyrilchampier** [DataDog/dd-trace-rb#1132]
Using gem version: ddtrace 0.38.0

Recently, we have this error message when launching our rspec suite on a rails project:
Full error message:  
```
W, [2020-08-03T13:29:43.057611 #15929]  WARN -- ddtrace: [ddtrace] Failed to collect environment information: undefined method `url' for #<Datadog::Transport::HTTP::Adapters::Test:0x00007fde4b256bc0> location: /Users/cyril/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/ddtrace-0.38.0/lib/ddtrace/diagnostics/environment_logger.rb:111:in `agent_url'
```

Here is our full configuration:
```
Datadog.configure do |c|
  c.use(:rails)

  c.tracer.enabled = true

  # Set transport to no-op mode. Enable tracer so that the stack is the same on every env (test, staging, etc.),
  # but only send the event to datadog agent in production.
  # The datadog agent will only accept apm http calls in production anyway.
  c.tracer.transport_options = proc { |t| t.adapter(:test) } unless Rails.env.production?
end
```

When I apply this patch:
```
module TestAdapterUrlMethod
  def url
    ''
  end
end
Datadog::Transport::HTTP::Adapters::Test.prepend(TestAdapterUrlMethod)
```

I get on stdout what seems to be a debug log with the warn level:
`W, [2020-08-03T13:30:38.668865 #16240]  WARN -- ddtrace: [ddtrace] DATADOG TRACER CONFIGURATION - {"date":"2020-08-03T13:30:38+02:00", ...`

**Comment by ericmustin** [comment]
Hey @cyrilchampier thanks for reaching out on this and reporting. I believe this is the same issue as https://github.com/DataDog/dd-trace-rb/issues/1125 , and we've made a PR this morning to address it https://github.com/DataDog/dd-trace-rb/pull/1131. I can ping on this issue when it's merged+released.

Regarding:
>`WARN -- ddtrace: [ddtrace] DATADOG TRACER CONFIGURATION - {"date":"2020-08-03T13:30:38+02:00"` ...

This is expected behavior from our startup logging, which was released in 0.38.0, and which can be disabled via env var `DD_TRACE_STARTUP_LOGS=false` or tracer setting `diagnostics.startup_logs.enabled`, You can read a bit more on it here: https://github.com/DataDog/dd-trace-rb/blob/master/docs/GettingStarted.md#tracer-settings

**Comment by cyrilchampier** [comment]
indeed, sorry for the duplicate!
